pkg/server/plugin/keymanager/awskms/awskms.go

```diff
@@ -131,11 +132,14 @@ func (p *Plugin) Configure(ctx context.Context, req *configv1.ConfigureRequest)
 		return nil, err
 	}
 
-	serverID, err := loadServerID(config.KeyMetadataFile)
-	if err != nil {
-		return nil, err
+	var serverID = config.KeyMetadata
+	if config.KeyMetadata == "" {
```

Comment:
This logic provides precedence to the `KeyMetadata` configuration over the `KeyMetadataFile`, allowing to have both configured, which looks confusing to me. I think that we should allow to have only one of the two. Please see my comment related with that.

Reply:
Makes sense. Fixed.